scenes/brick.gd

```diff
@@ -0,0 +1,15 @@
+extends StaticBody2D
+
+const simple_brick_scene = preload("res://scenes/power_up.tscn")
+
+func _ready():
+	pass
+
+func collapse_and_maybe_drop_power_up():
+	# This give us 10% of chance of a new PowerUp to be dropped
+	if (decision_var > 0.9):
```

Comment:
There is a :bug: here: the brick is only destroyed if it spawn a power-up.

Calling `queue_free()` should be done independently of spawning a power-up.